test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,394 @@
+    private static Component[] components;
+    private static boolean componentHidden = false;
+    private static boolean componentShown = false;
+    private static boolean componentMoved = false;
+    private static boolean componentResized = false;
+    private static ArrayList<ComponentEvent> events =
+        new ArrayList<ComponentEvent>();
+
+    private static final ComponentListener componentListener =
+        new ComponentListener() {
+
+            @Override
+            public void componentShown(ComponentEvent e) {
+                System.out.println("ComponentShown: " + e.getSource());
+                componentShown = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentResized(ComponentEvent e) {
+                System.out.println("ComponentResized: " + e.getSource());
+                componentResized = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+                System.out.println("ComponentMoved: " + e.getSource());
+                componentMoved = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+                System.out.println("ComponentHidden: " + e.getSource());
+                componentHidden = true;
+                events.add(e);
+            }
+        };
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
```

Comment:
I was notified robot.delay(1000); is more standard here

Reply:
i have make robot.delay(500); it works fine.